polkadot/network/src/lib.rs

```diff
@@ -420,10 +389,40 @@ impl PolkadotProtocol {
 			}
 			Message::BlockData(req_id, data) => self.on_block_data(ctx, peer_id, req_id, data),
 			Message::Collation(relay_parent, collation) => self.on_collation(ctx, peer_id, relay_parent, collation),
-			Message::CollatorRole(_) => {},
+			Message::CollatorRole(role) => self.on_new_role(ctx, peer_id, role),
 		}
 	}
 
+	fn on_session_key(&mut self, ctx: &mut Context<Block>, peer_id: PeerId, key: SessionKey) {
+		{
+			let info = match self.peers.get_mut(&peer_id) {
+				Some(peer) => peer,
+				None => return,
```

Comment:
This should be fairly uncommon, or? Us receiving Info about a peer we don't know about? I suspect the most common case is if the peer went down/disconnected before we received that message? While I agree we shouldn't fail in that case, I feel a bit uncomfortable with just dropping this without any info. Maybe put a `trace!` here for future race-condition-debugging?